scripts/lua/modules/alert_store/alert_store.lua

```diff
@@ -669,67 +669,173 @@ end
       label = severity_label,
       color = severity.color,
    }
 
    if tonumber(value["tstamp_end"]) > 0 and tonumber(value["tstamp"]) > 0 then
-      record["duration"] = tonumber(value["tstamp_end"]) - tonumber(value["tstamp"]) 
+      record[BASE_RNAME.DURATION.name] = tonumber(value["tstamp_end"]) - tonumber(value["tstamp"]) 
    elseif tonumber(value["tstamp"]) > 0 then
       local now = os.time()
-      record["duration"] = now - tonumber(value["tstamp"])
+      record[BASE_RNAME.DURATION.name] = now - tonumber(value["tstamp"])
    end
 
-   record["count"] = tonumber(value["count"]) or 1
+   record[BASE_RNAME.COUNT.name] = tonumber(value["count"]) or 1
 
    local alert_json = json.decode(value["json"]) or {}
-   record["script_key"] = alert_json["alert_generation"] and alert_json["alert_generation"]["script_key"]
+   record[BASE_RNAME.SCRIPT_KEY.name] = alert_json["alert_generation"] and alert_json["alert_generation"]["script_key"]
 
    return record
 end
 
+-- do not override in subclasses
+function alert_store:get_export_base_rnames()
+   return BASE_RNAME
+end
+
+-- to add new elements in subclasses define a RNAME table in subclass and returned it overring this function
+function alert_store:get_export_rnames()
+   return {}
+end
+
+-- do not override in subclasses
+function alert_store:get_export_record_names()
+   local columns_to_export = {}
+
+   for key, value in pairs(self:get_export_base_rnames()) do
+      if value.export then
+         columns_to_export[key] = value.name
+      end
+   end
+
+   for key, value in pairs(self:get_export_rnames()) do
+      if value.export then
+         columns_to_export[key] = value.name
+      end
+   end
+
+   return columns_to_export
+end
+
+-- Convert from table to CSV string
+function alert_store:to_csv(documents)
+
+   local separator = "|" -- change this to use another separator
+   local csv = ""
+
+   local csv_header = {} -- contains the column heading names
+   for _, value in pairs(self:get_export_record_names()) do
+      csv_header[#csv_header + 1] = tostring(value)
+   end
+   table.sort(csv_header) -- this assure the same csv columns order at each iteration
+
+   -- column heading output
+   local row = ""
+   for _, value in ipairs(csv_header) do
+      row = row .. separator .. self:escape_csv(value)
+   end
+   row = string.sub(row, 2) -- remove first separator
+   csv = csv .. row .. '\n'
+
+   if table.len(documents) > 0 then
+      -- csv row output
+      for _, document in ipairs(documents) do
+         row = ""
+         for _, column_name in ipairs(csv_header) do
+            row = row .. separator .. self:escape_csv(tostring(document[column_name]))
+         end
+         row = string.sub(row, 2) -- remove first separator
+         csv = csv .. row .. '\n'
+      end
+   end
+
+   return csv
+end
+
+-- Used to escape "'s by to_csv
+function alert_store:escape_csv(s)
+   if string.find(s, '[,"|]') then
+      s = '"' .. string.gsub(s, '"', '""') .. '"'
```

Comment:
this escape is not necessary for csv, it's the sqlite escape. Also note that this escape function doesn't work.